Account for changes in casacore master

VoronkovMA opened the pull request:
Recent changes in casacore/master (beyond released 3.4 version) made casarest not compilable again. This pull request fixes it.  

I started this work mainly because of two changes which break ASKAPsoft and, as it turned out, casarest builds in a number of places: 
  * casacore::Vector from std::vector constructor with is now explicit. Fixes are trivial and unlikely to cause any further issue.
  * type-related methods in measures. This is something I am less sure about but based on the code my change seems to be alright (and all tests pass). Quite possibly there was a bug with the wrong type used in casarest too (i.e. the same which prompted the change in casacore in the first place).

In addition, I have also removed some deprecation warnings related to some methods being moved from Table to TableUtil. However, I think it made the code not compilable against the released v3.4. The related changes are in a separate commit.

VoronkovMA commented:
And, I think, this TableUtil change trips the CI (presumably, it tries to build against the released casacore-3.4 rather than against the current head of master?).

VoronkovMA commented:
I have reverted the last commit (suppressing the warning about some methods moved to TableUtil), but kept it in history for future use. This way it will simplify things and my change should be compatible with both the last 3.4 release of casacore and the current HEAD of master branch.

aroffringa commented:
Looks like good fixes to me (including one change that should fix a bug -- curious that never got caught before).

I would indeed ignore the deprecation of Table functions after moving them to TableUtil for now, as it indeed would explicitly demand a recent casacore as you already mention. We do the same in e.g. Dp3 -- we'll ignore the warnings until the methods are definitely removed.

BTW I think you are one of the only users of casarest. We don't use it anywhere. If relevant, it might be interesting to strip out functionality that you use and get rid of parts that aren't used.

VoronkovMA commented:
We also use very little from casarest (I think just Components), perhaps some refactoring indeed can be done.

btw, who looks after releases? It may be good to have a casarest-1.8.1 released with this patch (it may be nice to release casacore as well to account for changes since v3.4 which was more than a year ago)?

VoronkovMA commented:
For future reference, changes related to Table vs TableUtil are stashed on my account-for-table-to-table-utils-methods-move branch of casarest.
